test/multidevice.cpp

```diff
@@ -93,12 +93,13 @@ void CommunicationTest::resetDstBuffers() {
 // It compares the given (possibly sharded) output with the result of the Fusion
 // run on a single device with the given (possibly sharded) inputs
 void PipelineTest::validate() {
-  // execute the fusion on one device without pipeline scheduling
-  auto fusion_copy = std::make_unique<Fusion>(*runtime->completeFusion());
-  unshard(fusion_copy.get());
-  FusionExecutorCache unsharded_fec(std::move(fusion_copy));
-  auto ref_unsharded_outputs =
-      unsharded_fec.runFusionWithInputs(unsharded_inputs);
+  if (ref_unsharded_outputs.empty()) {
```

Comment:
This adds complexity because ref_unsharded_outputs may be filled outside this function and whether it's filled affects the logic inside the function. Is it possible to avoid this added complexity in this PR?

Reply:
porting to other pr:
https://github.com/NVIDIA/Fuser/pull/1925/files#r1523869069